### ISSUE TITLE
One-Click Postage always returns OCP01 on Safari

### ISSUE THREAD

**Issue by AprilSylph** [new-xkit/XKit#1540]
from the Discord and the askbox: numerous reports of OCP not working on Firefox and Safari

able to reproduce on FF59.0.2 on Linux (elementary OS)
report from Discord was on FF ESR 52.7.3 on MacOS 10.9.5
#1541 reports FF59.0.2 on Win10

cause as of yet unknown. could be affecting other extensions too.

P-Critical because OCP is one of _the_ most-used extensions and it's broken on two entire browsers

**Comment by roblprince** [comment]
Can confirm this issue has appeared in Safari as well (11.1) and also using a Mac OS (10.13.4).

**Comment by insectoid5** [comment]
Can confirm the Firefox issue.  OCP was working fine just this morning, then was broken this afternoon.

**Comment by AprilSylph** [comment]
Interestingly, the response ([text](https://gist.github.com/AprilSylph/33caa374d2e5a4b385945e5c8a88da1d)/[html](https://workbench.sylph.tech/error) says that Tumblr engineers have been notified, and the `data-status-code` is a very vague and unuseful 500. This would explain why completely different versions of Firefox are suddenly failing with no updates, but it still doesn't make complete sense to me.

**Comment by nightpool** [comment]
case sensitivity issue again maybe?

**Comment by AprilSylph** [comment]
confirmed that this happens for unupdated browsers; just woke up my laptop from a 2.5 month hibernation, didn't refresh anything (had a tumblr tab already open, typical), OCP fails. what I want now is an explanation of why the hell Chrome is fine

---
@nightpool i don't _think_ so? OCP's `post` function doesn't use kitties at all, and the json it does use is exactly the same on chrome and FF. considering this is happening without any user-side updates, this has to be tumblr being weird, surely?

**Comment by hobinjk** [comment]
Okay so the failing request for me is /svc/post/fetch. The difference between ffx and chrome is the data provided in the post which is as follows:
firefox
```javascript
{
  "form_key":"pljTumXOVyciZH1sfLTwuL0vBk",
  "channel_id":"thezed",
  "detached":true,
  "reblog":true,
  "reblog_id":173276293079,
  "reblog_key":"ILx6mHTY",
  "errors":false,
  "created_post":true,
  "context_page":"dashboard",
  "post_context_page":"dashboard",
  "silent":false,
  "context_id":"",
  "reblog_post_id":"173276293079",
  "is_rich_text[one]":"0",
  "is_rich_text[two]":"1",
  "is_rich_text[three]":"0",
  "post[slug]":"",
  "post[draft_status]":"",
  "post[date]":"",
  "post[type]":"photo",
  "post[tags]":"",
  "post[two]":"",
  "post[publish_on]":"",
  "post[state]":1,
  "custom_tweet":""
}
```
chrome
```javascript
{
  "channel_id":"heart",
  "reblog_id":173276293079,
  "reblog_key":"ILx6mHTY",
  "form_key":"pljTumXOVyciZH1sfLTwuL0vBk",
  "post_type":"photoset"
}
```

**Comment by hobinjk** [comment]
tl;dr: Accept-Encoding so I'm guessing a route is setting some Vary wrong headers

**Comment by hobinjk** [comment]
Nevermind, it's the Origin/Referer headers because I can't read (and that actually makes sense)

**Comment by AprilSylph** [comment]
re: safari, would it be possible to use `add_function` (with a regular XMLHttpRequest and custom events to communicate with the extension) to fool the browser into sending the right headers?
(i'm gonna try it anyway, but if anyone would like to explain why it'll never work, please do)

**Comment by AprilSylph** [comment]
the XHR via `add_function` method seems to work, although custom events seem not to - postMessage is fine though

**Comment by nightpool** [comment]
why was this reopened? did I miss something?

**Comment by nightpool** [comment]
ah, safari
